torch/fx/symbolic_trace.py

```diff
@@ -307,13 +361,52 @@ def forward(*args, **kwargs):
             # Seems to be a mypy limitation: https://github.com/python/mypy/issues/2427
             torch.nn.Module.__getattr__ = module_getattr_wrapper  # type: ignore
             torch.nn.Module.__call__ = module_call_wrapper
+
+            orig_fns = self._patch_wrapped_functions()
+
             self.create_node('output', 'output', (self.create_arg(fn(*args)),), {},
                              type_expr=fn.__annotations__.get('return', None))
         finally:
+            self._unpatch_wrapped_functions(orig_fns)
             torch.nn.Module.__call__ = orig_call
             torch.nn.Module.__getattr__ = orig_getattr  # type: ignore
         return self.graph
 
+# Type Dict['frame', Dict['str', Any]]
+# (can't annotate this global for some reason)
+#
+# This maps from frame object to ordered set of function names
+# to be patched at the beginning of symbolic tracing.
+_wrapped_fn_patch_table = {}
+
+def wrap(fn_name : str):
+    """
+    This function can be called at global scope in a module to cause
+    references to the global function secified by `fn_name` to use
+    them in FX.
+
+        # foo/bar/baz.py
+        def my_custom_function(x, y):
+            return x * x + y * y
+
+        torch.fx.wrap('my_custom_function')
+
+        def fn_to_be_traced(x, y):
+            # When symbolic tracing, the below call to my_custom_function will be inserted into
+            # the graph rather than tracing it.
+            return my_custom_function(x, y)
+
+    Args:
+
+        fn_name (str): The name of the global function to insert into the graph when it's called
+    """
+    f = inspect.currentframe().f_back
+    if f.f_code.co_name != '<module>':
+        raise NotImplementedError('wrap must be called at the top level of a module')
+
+    global _wrapped_fn_patch_table
+    _wrapped_fn_patch_table.setdefault(f, {})
```

Comment:
It's not a good idea to capture the frame here, it holds `f_back` references to the entire import tree of the file and will probably cause a bunch of objects to stay reachable. Instead, we probably just want to record something like tuples of `(global_dictionary, symbol)`